packages/gradle/src/utils/get-gradle-report.ts

```diff
@@ -73,44 +58,22 @@ export async function populateGradleReport(
   const gradleProjectReportStart = performance.mark(
     'gradleProjectReport:start'
   );
-  let projectReportLines;
-  try {
-    projectReportLines = await execGradleAsync(['projectReportAll'], {
-      cwd: workspaceRoot,
-    });
-  } catch (e) {
-    try {
-      projectReportLines = await execGradleAsync(['projectReport'], {
-        cwd: workspaceRoot,
-      });
-      logger.warn(
-        'Could not run `projectReportAll` task. Ran `projectReport` instead. Please run `nx generate @nx/gradle:init` to generate the necessary tasks.'
-      );
-    } catch (e) {
-      throw new AggregateCreateNodesError(
-        [
-          [
-            null,
-            new Error(
-              'Could not run `projectReportAll` or `projectReport` task. Please run `nx generate @nx/gradle:init` to generate the necessary tasks.'
-            ),
-          ],
-        ],
-        []
-      );
-    }
-  }
-  projectReportLines = projectReportLines
-    .toString()
-    .split(newLineSeparator)
-    .filter((line) => line.trim() !== '');
+
+  const projectReportLines: string[] = (
+    await Promise.all(
+      gradlewFiles.map(
+        async (gradlewFile) => await getProjectReportLines(gradlewFile)
+      )
+    )
```

Comment:
Test if this is faster synchronously or async. There's a chance that gradle does not like being run in parallel with itself because it parallelizes across multiple processes.

Reply:
i tried on a project with mutiple gradle app, it is slightly faster synchronously